x/utils/key_tagger_test.go

```diff
@@ -131,3 +133,56 @@ func TestKeyTagger(t *testing.T) {
 		})
 	}
 }
+
+// writeDecorator writes the key, value pair.
+// either before or after calling the handlers
+type writeDecorator struct {
+	key   []byte
+	value []byte
+	after bool
+}
+
+var _ weave.Decorator = writeDecorator{}
+
+func (d writeDecorator) Check(ctx weave.Context, store weave.KVStore, tx weave.Tx, next weave.Checker) (weave.CheckResult, error) {
+	if !d.after {
+		store.Set(d.key, d.value)
+	}
+	res, err := next.Check(ctx, store, tx)
+	if d.after && err == nil {
+		store.Set(d.key, d.value)
+	}
+	return res, err
+}
+
+func (d writeDecorator) Deliver(ctx weave.Context, store weave.KVStore, tx weave.Tx, next weave.Deliverer) (weave.DeliverResult, error) {
+	if !d.after {
+		store.Set(d.key, d.value)
+	}
+	res, err := next.Deliver(ctx, store, tx)
+	if d.after && err == nil {
+		store.Set(d.key, d.value)
+	}
+	return res, err
+}
+
+// tagHandler writes the key, value pair and returns the error (may be nil)
```

Comment:
Can't you implement this with a properly constructed 
`weavetest.Handler{DeliverResult: weave.DeliverResult{Tags:  common.KVPairs{{Key: h.key, Value: h.value}}}}`

sure, reads ugly, but maybe a helper function here rather than writing a struct?

Reply:
I have copied this without reading the implementation :see_no_evil: I will look into this and update if possible.